packages/sdk/src/types/runtime/actor.ts

```diff
@@ -311,59 +314,67 @@ export class Actor implements ActorLike, Patchable<ActorLike> {
-    // ): ForwardPromise<Collider> {
-    //     if (!this._collider) {
-    //         this._collider = new Collider(this);
-    //         observe(this._collider, 'collider', (...path: string[]) => this.actorChanged(...path));
-    //         this.subscribe('collider');
-    //         return this.context.internal.enableCollider(
-    //             this.id,
-    //             colliderType,
-    //             collisionLayer,
-    //             isTrigger,
-    //             center,
-    //             size);
-    //     }
-    //     return ForwardPromise.Resolve(this._collider);
-    // }
+    /**
+     * Adds a collider of the given type and parameters on the actor.
+     * @param colliderType Type of the collider to enable.
+     * @param collisionLayer The layer that the collider operates in.
+     * @param isTrigger Whether the collider is a trigger volume or not.
+     * @param center The center of the collider, or default of the object if none is provided.
+     * @param radius The radius of the collider, or default bounding if non is provided.
+     */
+    public enableCollider(
+        colliderType: 'sphere',
+        collisionLayer: CollisionLayer,
+        isTrigger: boolean,
+        center?: Vector3Like,
+        radius?: number): void;
+
+    /**
+     * Adds a collider of the given type and parameters on the actor.
+     * @param colliderType Type of the collider to enable.
+     * @param collisionLayer The layer that the collider operates in.
+     * @param isTrigger Whether the collider is a trigger volume or not.
+     * @param center The center of the collider, or default of the object if none is provided.
+     * @param size
+     */
+    public enableCollider(
+        colliderType: 'box',
+        collisionLayer: CollisionLayer,
+        isTrigger: boolean,
+        center?: Vector3Like,
+        size?: Vector3Like): void;
+    
+    /** @ignore */
+    public enableCollider(
+        colliderType: 'box' | 'sphere',
+        collisionLayer: CollisionLayer,
+        isTrigger: boolean,
+        center?: Vector3Like,
+        size?: number | Vector3Like
+    ): void {
+        if (!this._collider) {
+            this._collider = new Collider(this);
+            observe({
+                target: this._collider,
+                targetName: 'collider',
+                notifyChanged: (...path: string[]) => this.actorChanged(...path),
+                // Trigger notifications for every observed leaf node to ensure we get all values in the initial patch.
+                triggerNotificationsNow: true
+            });
+
+            // Copy to the _collider
+            const colliderParams = this._generateColliderParams(colliderType, center, size);
+
+            if (colliderParams) {
+                this._collider.copy({
+                    enabled: true,
+                    isTrigger,
+                    collisionLayer,
+                    colliderParams
+                } as ColliderLike)
+            }

```

Comment:
The other components unconditionally copy the values (regardless of whether the component existed already). Would be good to do that here to, for consistency. Is there a reason not to?

Reply:
The collider params are immutable, so they shouldn't be set if there already exists a collider.  We had discussed this the other day.

Reply:
Ok. I didn't realize all of them were immutable.

---
In reply to: [257434402](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/203#discussion_r257434402) [](ancestors = 257434402)